src/core/dbus-execute.c

```diff
@@ -3140,6 +3141,36 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
+        } else if (streq(name, "ExecSearchPath")) {
+                _cleanup_strv_free_ char **l = NULL;
+                char **p;
+
+                r = sd_bus_message_read_strv(message, &l);
+                if (r < 0)
+                        return r;
+
+                STRV_FOREACH(p, l) {
+                        if (!path_is_absolute(*p) || strchr(*p, ':'))
```

Comment:
please match up this validation with the one your doing in config_parse_colon_separated_paths(), i.e. path_is_normalized() is missing here.